src/Hydrodynamics/Hydrodynamics.cpp

```diff
@@ -123,6 +124,67 @@ Hydrodynamics<ndim>::Hydrodynamics(int hydro_forces_aux, int self_gravity_aux, F
 
 
 //=================================================================================================
+//  Hydrodynamics::CreateNewParticle
+/// Create a new hydro particle in the main arrays at the end of the existing particles.
+/// Also triggers the re-creation of ghosts and another tree-build to include new particles.
+/// Needs to be called soon before the load balancing and tree rebuild because it will invalidate
+/// the tree and the ghosts
+//=================================================================================================
+template <int ndim>
+Particle<ndim>& Hydrodynamics<ndim>::CreateNewParticle
+ (const enum ptype _ptype,             ///< [in] ptype of new particle
+  const enum parttype _parttype,       ///< [in] parttype of new particle
+  const int n,                         ///< [in] Current integer time
+  const int level_step,                ///< [in] ..
+  const int level_max,                 ///< [in] Current maximum block timestep level
+  const FLOAT t,                       ///< [in] Current simulation time
+  const FLOAT m,                       ///< [in] Mass of new particle
+  const FLOAT u,                       ///< [in] Specific internal energy of new particle
+  const FLOAT r[ndim],                 ///< [in] Position of new particle
+  const FLOAT v[ndim],                 ///< [in] Velocity of new particle
+  SimulationBase* sim)				   ///< [inout] Simulation object
+{
```

Comment:
I partly agree with this.  Looking at it now, a lot of this info can simply be taken from the simulation object (using the sim object pointer).  However, the intention of including this other info in the function interface is to force the user to give the minimum required amount of information needed to construct the particle in memory without creating problems (all other quantities important can then be derived from this).  If this wasn't the case, the user might for example only set r and m and forget about u and v and then cause grief elsewhere.  And even if they remembered these four, they might forget (or not even know) to also set r0, v0, etc..  So, I'm in favour of keeping this information in the function interface since I think it's more robust.  I will however remove those other unnecessary things now.